packages/pretty-format/src/plugins/lib/printImmutable.js

```diff
@@ -28,19 +28,29 @@ const printImmutable = (
   isMap: boolean,
 ): string => {
   const [openTag, closeTag] = isMap ? ['{', '}'] : ['[', ']'];
+  const fullStructureName = val._name || immutableDataStructureName;
+
   let result =
     IMMUTABLE_NAMESPACE +
-    immutableDataStructureName +
+    fullStructureName +
     SPACE +
     openTag +
     opts.edgeSpacing;
 
   const immutableArray = [];
-  val.forEach((item, key) =>
+
+  const pushToImmutableArray = (item: any, key: string) => {
     immutableArray.push(
       indent(addKey(isMap, key) + print(item, print, indent, opts, colors)),
-    ),
-  );
+    );
+  };
+
+  if (Array.isArray(val._keys)) {
+    // if we have a record, we can not iterate on it directly
+    val._keys.forEach(key => pushToImmutableArray(val.get(key), key));
```

Comment:
We are looking at private fields here and with `._name` above. I'm worried about changes in Immutable that will break this and therefore break snapshots.

cc @leebyron do you have any input for us on what we should do here? Do you think this is fine for pretty-printing?